src/libstd/sys/unix/process.rs

```diff
@@ -254,6 +259,81 @@ impl Process {
             unsafe { libc::_exit(1) }
         }
 
+        fn walk_open_fd_brute<F>(action_fd: F) where F: Fn(RawFd) {
+            // The `getdtablesize(3)` and `sysconf(_SC_OPEN_MAX)` could miss file descriptors if
+            // the RLIMIT_NOFILE is lowered (cf. #12148) but there is no good way to list them all
+            // except with procfs (cf. `walk_open_fd_proc()`).
+            // However, an unprivileged process (i.e. without CAP_SYS_RESOURCE on Linux) is not
+            // allowed to use a file descriptor upper or equal to it's hard limit nor to change
+            // this limit upwards.
+            let begin = libc::STDERR_FILENO + 1;
+            let end = unsafe { libc::sysconf(libc::consts::os::sysconf::_SC_OPEN_MAX) } as RawFd;
```

Comment:
This is also wrong.  Both `getdtablesize` and `_SC_OPEN_MAX` are for the maximum number of file descriptors a process can allocate.  This is not the same as the amount of file descriptors one already has.  It is entirely possible for a program to have leaked in a high file descriptor to your process and also have your process be restricted to a tiny amount of file descriptors.